File: CS/GridViewVisibleCellsHelper.cs
using System;
using System.Collections.Generic;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Views.Grid.ViewInfo;
using DevExpress.XtraGrid.Columns;
using System.Drawing;

namespace WindowsApplication1
{
    public static class GridViewVisibleCellsHelper
    {

        public static GridViewInfo GetViewInfo(GridView view)
        {
            return view.GetViewInfo() as GridViewInfo;
        }
        public static int GetTopVisibleRow(this GridView view)
        {
            return view.TopRowIndex;
        }

        public static int GetVisibleRowCount(this GridView view)
        {
            return GetViewInfo(view).RowsLoadInfo.ResultRowCount;
        }
        public static int GetBottomVisibleRow(this GridView view)
        {
            int visibleRowsCount = GetVisibleRowCount(view);
            return view.GetTopVisibleRow() + visibleRowsCount - 1;
        }

        public static GridColumn GetFirstVisibleColumn(this GridView view)
        {
            return GetViewInfo(view).ColumnsInfo.FirstColumnInfo.Column;
        }

        public static GridColumn GetLastVisibleColumn(this GridView view)
        {
            return GetViewInfo(view).ColumnsInfo.LastColumnInfo.Column;
        }

        public static Rectangle GetVisibleCells(this GridView view)
        {
            int x = view.GetFirstVisibleColumn().VisibleIndex;
            int y = view.GetTopVisibleRow();
            int width = view.GetLastVisibleColumn().VisibleIndex - x;
            int height = view.GetBottomVisibleRow() - y;
            return new Rectangle(x, y, width, height);
        }

        public static Rectangle GetAllCellsArea(this GridView view)
        {
            int x = 0;
            int y = 0;
            int width = view.VisibleColumns.Count;
            int height =view.RowCount;
            return new Rectangle(x, y, width, height);
        }
    }
}

File: VB/GridViewVisibleCellsHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Views.Grid.ViewInfo
Imports DevExpress.XtraGrid.Columns
Imports System.Drawing

Namespace WindowsApplication1
	Public  Module GridViewVisibleCellsHelper

		
		Public  Function GetViewInfo(ByVal view As GridView) As GridViewInfo
			Return TryCast(view.GetViewInfo(), GridViewInfo)
		End Function
		<System.Runtime.CompilerServices.Extension> _
		Public  Function GetTopVisibleRow(ByVal view As GridView) As Integer
			Return view.TopRowIndex
		End Function

		<System.Runtime.CompilerServices.Extension> _
		Public  Function GetVisibleRowCount(ByVal view As GridView) As Integer
			Return GetViewInfo(view).RowsLoadInfo.ResultRowCount
		End Function
		<System.Runtime.CompilerServices.Extension> _
		Public  Function GetBottomVisibleRow(ByVal view As GridView) As Integer
			Dim visibleRowsCount As Integer = GetVisibleRowCount(view)
			Return view.GetTopVisibleRow() + visibleRowsCount - 1
		End Function

		<System.Runtime.CompilerServices.Extension> _
		Public  Function GetFirstVisibleColumn(ByVal view As GridView) As GridColumn
			Return GetViewInfo(view).ColumnsInfo.FirstColumnInfo.Column
		End Function

		<System.Runtime.CompilerServices.Extension> _
		Public  Function GetLastVisibleColumn(ByVal view As GridView) As GridColumn
			Return GetViewInfo(view).ColumnsInfo.LastColumnInfo.Column
		End Function

		<System.Runtime.CompilerServices.Extension> _
		Public  Function GetVisibleCells(ByVal view As GridView) As Rectangle
			Dim x As Integer = view.GetFirstVisibleColumn().VisibleIndex
			Dim y As Integer = view.GetTopVisibleRow()
			Dim width As Integer = view.GetLastVisibleColumn().VisibleIndex - x
			Dim height As Integer = view.GetBottomVisibleRow() - y
			Return New Rectangle(x, y, width, height)
		End Function

		<System.Runtime.CompilerServices.Extension> _
		Public  Function GetAllCellsArea(ByVal view As GridView) As Rectangle
			Dim x As Integer = 0
			Dim y As Integer = 0
			Dim width As Integer = view.VisibleColumns.Count
			Dim height As Integer =view.RowCount
			Return New Rectangle(x, y, width, height)
		End Function
	End Module
End Namespace
